pkgs/by-name/fl/flatpak/package.nix

```
@@ -89,43 +103,53 @@ stdenv.mkDerivation (finalAttrs: {
     hash = "sha256-ywrFZa3LYhJ8bRHtUO5wRNaoNvppw1Sy9LZAoiv6Syo=";
   };
 
-  patches = [
-    # Hardcode paths used by tests and change test runtime generation to use files from Nix store.
-    # https://github.com/flatpak/flatpak/issues/1460
-    (replaceVars ./fix-test-paths.patch {
-      inherit
-        coreutils
-        gettext
-        gtk3
-        socat
-        ;
-      dfu = desktop-file-utils;
-      hicolorIconTheme = hicolor-icon-theme;
-      smi = shared-mime-info;
-    })
-
-    # Hardcode paths used by Flatpak itself.
-    (replaceVars ./fix-paths.patch {
-      p11kit = lib.getExe p11-kit;
-    })
+  patches =
+    [
+      # Use flatpak from PATH to avoid references to `/nix/store` in `/desktop` files.
+      # Applications containing `DBusActivatable` entries should be able to find the flatpak binary.
+      # https://github.com/NixOS/nixpkgs/issues/138956
+      ./binary-path.patch
 
-    # Use flatpak from PATH to avoid references to `/nix/store` in `/desktop` files.
-    # Applications containing `DBusActivatable` entries should be able to find the flatpak binary.
-    # https://github.com/NixOS/nixpkgs/issues/138956
-    ./binary-path.patch
+      # Try mounting fonts and icons from NixOS locations if FHS locations don't exist.
+      # https://github.com/NixOS/nixpkgs/issues/119433
+      ./fix-fonts-icons.patch
 
-    # Try mounting fonts and icons from NixOS locations if FHS locations don't exist.
-    # https://github.com/NixOS/nixpkgs/issues/119433
-    ./fix-fonts-icons.patch
+      # Nix environment hacks should not leak into the apps.
+      # https://github.com/NixOS/nixpkgs/issues/53441
+      ./unset-env-vars.patch
 
-    # Nix environment hacks should not leak into the apps.
-    # https://github.com/NixOS/nixpkgs/issues/53441
-    ./unset-env-vars.patch
+      # The icon validator needs to access the gdk-pixbuf loaders in the Nix store
+      # and cannot bind FHS paths since those are not available on NixOS.
+      finalAttrs.passthru.icon-validator-patch
 
-    # The icon validator needs to access the gdk-pixbuf loaders in the Nix store
-    # and cannot bind FHS paths since those are not available on NixOS.
-    finalAttrs.passthru.icon-validator-patch
-  ];
+      (fetchpatch {
+        name = "static.patch";
+        url = "https://github.com/flatpak/flatpak/commit/114c22e814fc28243585915321b8e943471c377f.patch";
+        hash = "sha256-3JLzG74myBTssXQau0Ei5rpthy93Va7xb2MHRnJ3kaI=";
+      })
+    ]
+    ++ lib.optionals finalAttrs.doCheck [
```

```suggestion
    ++ lib.optionals finalAttrs.finalPackage.doCheck [
```
This way we can ensure `stdenv.buildPlatform.canExecute stdenv.hostPlatform` is always true

---

I actually noticed this was wrong for the installedTests output, but wanted to limit unrelated changes.  I can fix it though.

---

Done.

---

I've changed it back because using `finalAttrs.finalPackage.doCheck` in `outputs` causes an infinite recursion, which is presumably why it was done that way in the first place.